File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        int size = s.length(), count = 0;

        for(int i = 0 ; i < k ; i++) {
            if(isVowel(s.charAt(i))) count++;
        }

        int maxCount = count;

        for(int i = k ; i < size ; i++) {
            if(isVowel(s.charAt(i))) count++;
            if(isVowel(s.charAt(i - k))) count--;

            maxCount = Math.max(maxCount, count);
        }

        return maxCount;
    }

    public boolean isVowel(char c) {
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
    }
}

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        size, count = len(s), 0

        for i in range(k):
            if self.isVowel(s[i]): 
                count += 1
        
        maxCount = count

        for i in range(k, size):
            if self.isVowel(s[i]): 
                count += 1
            if self.isVowel(s[i - k]):
                count -= 1

            maxCount = max(maxCount, count)

        return maxCount

    def isVowel(self, c):
        return c == 'a' or c == 'e' or c == 'i' or c == 'o' or c == 'u'